micromanager_gui/_core_widgets/_presets_widget.py

```diff
@@ -0,0 +1,128 @@
+from typing import Optional, Tuple
+
+from pymmcore_plus import DeviceType
+from qtpy.QtWidgets import QComboBox, QHBoxLayout, QWidget
+from superqt.utils import signals_blocked
+
+from .._core import get_core_singleton
+from .._util import _set_wdg_color
+
+
+class PresetsWidget(QWidget):
+    """Create a QCombobox Widget for a specified group presets"""
+
+    def __init__(
+        self,
+        group: str,
+        text_color: str = "black",
+        parent: Optional[QWidget] = None,
+    ) -> None:
+
+        super().__init__(parent)
+
+        self._mmc = get_core_singleton()
+
+        self._group = group
+        self.text_color = text_color
+
+        if self._group not in self._mmc.getAvailableConfigGroups():
+            raise ValueError(f"{self._group} group does not exist.")
+
+        self._presets = list(self._mmc.getAvailableConfigs(self._group))
+
+        if not self._presets:
+            raise ValueError(f"{self._group} group does not have presets.")
+
+        self._combo = QComboBox()
+        self._combo.addItems(self._presets)
+
+        self.setLayout(QHBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(self._combo)
+        _set_wdg_color(self.text_color, self._combo)
+        self._combo.currentTextChanged.connect(self._on_combo_changed)
+        self._combo.textActivated.connect(self._on_text_activate)
+
+        self._mmc.events.configSet.connect(self._on_cfg_set)
+        self._mmc.events.configSet.connect(self._highlight_if_preset_state_changed)
```

Comment:
This should be changed to be `configChanged` and check whether we are in a valid config state.